File: app/src/main/java/com/bridou_n/beaconscanner/dagger/ContextModule.java
package com.bridou_n.beaconscanner.dagger;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ContextModule {

    private final Application context;

    public ContextModule(Application app) {
        context = app;
    }

    @Provides @Singleton
    public Context providesContext() {
        return context;
    }
}

File: app/src/main/java/com/bridou_n/beaconscanner/dagger/ContextModule.kt
package com.bridou_n.beaconscanner.dagger

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ContextModule(private val context: Application) {

    @Provides @Singleton
    fun providesContext() : Context = context
}
